### PR TITLE
feat: migrate extension key

### PR DESCRIPTION
**Pull request by antsgar**

### CODE REVIEW

File: packages/snjs/lib/migrations/2_7_23.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { Migration } from '@Lib/migrations/migration';
+import { ContentType } from '@Lib/models';
+import { SettingName } from '@standardnotes/settings';
+
+export class Migration2_7_23 extends Migration {
+  static version(): string {
+    return '2.7.23';
+  }
+
+  protected registerStageHandlers(): void {
```
Comment (mobitar) [inline packages/snjs/lib/migrations/2_7_23.ts:11]: Sign in event is only on explicit sign in. And actually you don’t have to be signed in to have a repo. This should be on full sync completed instead.

**Comment by mobitar** [comment]
Is this testable?

File: packages/snjs/lib/migrations/2_7_23.ts (hunk 1 above)
Comment (antsgar) [reply packages/snjs/lib/migrations/2_7_23.ts:11]: But we need a user uuid to be able to update the user setting

Comment (mobitar) [reply packages/snjs/lib/migrations/2_7_23.ts:11]: Ah ok, I’m that case on full sync completed and if signed in.

**Comment by antsgar** [comment]
> Is this testable?

It should be, let me give it a try 🙂

**Comment by antsgar** [comment]
@mobitar added the extension repo items observer, but I suppose we should still keep the migration for already existing items?

**Comment by antsgar** [comment]
Depends on #385 since the observer has been added to the new features service

**Review by karolsojko — APPROVED** [review]
Good stuff - glad to know we have these kind of mechanisms in place 👍

**Comment by antsgar** [comment]
Closing in favor of #394 (PR shows wrong commits after dependent PR was squash-merged)

**Pull request closed without merging**